packages/dds/cell/src/test/cell.rollback.spec.ts

```diff
@@ -0,0 +1,166 @@
+		containerRuntime.flush();
+		containerRuntimeFactory.processAllMessages();
+
+		cell.delete();
+		assert(cell.empty());
+
+		// rollback delete
+		containerRuntime.rollback?.();
+		assert.equal(cell.get(), 42);
+
+		// delete triggers delete event, rollback restores valueChanged
+		assert.deepEqual(events, ["valueChanged", "delete", "valueChanged"]);
+	});
+});
+
+describe("SharedCell rollback events with multiple clients", () => {
+	it("should emit valueChanged on set and rollback should re-emit previous value across clients", async () => {
+		// Setup two clients
+		const {
+			dds: cell1,
+			containerRuntimeFactory,
+			containerRuntime: runtime1,
+		} = setupRollbackTest<ISharedCell>(
+			"client-1",
+			(rt, id): ISharedCell => cellFactory.create(rt, id),
+		);
+		const { dds: cell2 } = createAdditionalClient(
+			containerRuntimeFactory,
+			"client-2",
+			(rt, id): ISharedCell => cellFactory.create(rt, `cell-${id}`),
+		);
+
+		const events1: string[] = [];
+		const events2: string[] = [];
+
+		// Listen to valueChanged events on both clients
+		cell1.on("valueChanged", () => events1.push("valueChanged"));
+		cell2.on("valueChanged", () => events2.push("valueChanged"));
+
+		// Client 1 sets a value
+		cell1.set(10);
+		assert.equal(cell1.get(), 10);
+
+		// Propagate ops to client 2
+		runtime1.flush();
+		containerRuntimeFactory.processAllMessages();
+
+		assert.equal(cell2.get(), 10);
+
+		cell1.set(100);
+		assert.equal(cell1.get(), 100);
+		assert.equal(cell2.get(), 10);
+
+		// Rollback on client 1
+		runtime1.rollback?.();
+
+		assert.equal(cell1.get(), 10);
+		assert.equal(cell2.get(), 10);
+
+		// Both clients should have seen the events
+		assert.deepEqual(events1, ["valueChanged", "valueChanged", "valueChanged"]);
+		assert.deepEqual(events2, ["valueChanged"]);
+	});
+
+	it("should emit delete on delete, and rollback should re-emit last valueChanged across clients", async () => {
+		// Setup two clients
+		const {
+			dds: cell1,
+			containerRuntimeFactory,
+			containerRuntime: runtime1,
+		} = setupRollbackTest<ISharedCell>(
+			"client-1",
+			(rt, id): ISharedCell => cellFactory.create(rt, id),
+		);
+		const { dds: cell2 } = createAdditionalClient(
+			containerRuntimeFactory,
+			"client-2",
+			(rt, id): ISharedCell => cellFactory.create(rt, `cell-${id}`),
+		);
+
+		const events1: string[] = [];
+		const events2: string[] = [];
+
+		// Attach listeners
+		cell1.on("valueChanged", () => events1.push("valueChanged"));
+		cell1.on("delete", () => events1.push("delete"));
+
+		cell2.on("valueChanged", () => events2.push("valueChanged"));
+		cell2.on("delete", () => events2.push("delete"));
+
+		// Set initial value and propagate
+		cell1.set(42);
+		runtime1.flush();
+		containerRuntimeFactory.processAllMessages();
+
+		assert.equal(cell2.get(), 42);
+
+		// Delete the value
+		cell1.delete();
+
+		assert(cell1.empty());
+		assert.equal(cell2.get(), 42);
+
+		// Rollback delete
+		runtime1.rollback?.();
```

Comment:
Would probably be good to do a final flush/process (that we expect to be a no-op) to verify nothing slips out that ends up in cell2/events2.
```suggestion
		runtime1.rollback?.();
		// After rollback, this flush/process should not affect cell2.
		runtime1.flush();
		containerRuntimeFactory.processAllMessages();
```